### ISSUE TITLE
file_exists endpoint needs to check sanitized name for folders

### ISSUE THREAD

**Issue by susannemoog** [TYPO3-Initiatives/digital-asset-management#77]
Currently the file_exists endpoint returns false if a folder was renamed after upload - it's working for files on top level. Examples: 

file named: "2019_01_29 10_13 AM Office Lens.jpg" -> "2019_01_29_10_13_AM_Office_Lens.jpg" after upload, second upload of the same file reports "file exists"

folder named "1. Mai" becomes "1._Mai" after upload. Uploading the same folder again, the file_exists check returns false for all files in there, but the upload fails as the folder exists and can't be created again (and the override check is also missing then).

**Comment by lolli42** [comment]
yeah. our fileExists strategy does not work here since fal does not sanitize identifiers internally, but the identifier is created client side without being properly sanitized when uploading / testing for existance. we'll have to rewrite both methods to do that properly ourselfs on server side.
